pallets/gear-built-in-actors/src/lib.rs

```diff
@@ -0,0 +1,443 @@
+            .try_into()
+            .unwrap_or_else(|_| unreachable!("Response message is too large"));
+        let reply_id = MessageId::generate_reply(message_id);
+        let packet = ReplyPacket::new(payload, 0);
+        let dispatch =
+            ReplyMessage::from_packet(reply_id, packet).into_dispatch(actor_id, origin, message_id);
+
+        journal.push(JournalNote::SendDispatch {
+            message_id,
+            dispatch,
+            delay: 0,
+            reservation: None,
+        });
+
+        let outcome = DispatchOutcome::Success;
+        journal.push(JournalNote::MessageDispatched {
+            message_id,
+            source: origin,
+            outcome,
+        });
+
+        journal.push(JournalNote::MessageConsumed(message_id));
+
+        journal
+    }
+
+    // Error in the actor, generates error reply
+    fn process_error(dispatch: &StoredDispatch, err: BuiltInActorReason) -> Vec<JournalNote> {
+        let message_id = dispatch.id();
+        let origin = dispatch.source();
+        let actor_id = dispatch.destination();
+
+        let mut journal = vec![];
+
+        // No call dispatced, so no gas burned except for the base
+        journal.push(JournalNote::GasBurned {
+            message_id,
+            amount: BASE_GAS_BURNED,
+        });
+
+        let err_payload = err
+            .to_string()
+            .into_bytes()
+            .try_into()
+            .unwrap_or_else(|_| unreachable!("Error message is too large"));
+        let err: SimpleExecutionError = err.into();
+
+        let dispatch = ReplyMessage::system(message_id, err_payload, err)
+            .into_dispatch(actor_id, origin, message_id);
+
+        journal.push(JournalNote::SendDispatch {
+            message_id,
+            dispatch,
+            delay: 0,
+            reservation: None,
+        });
+
+        let outcome = DispatchOutcome::MessageTrap {
+            program_id: actor_id,
+            trap: err.to_string(),
+        };
+        journal.push(JournalNote::MessageDispatched {
+            message_id,
+            source: origin,
+            outcome,
+        });
+
+        journal.push(JournalNote::MessageConsumed(message_id));
+
+        journal
+    }
+
+    fn check_gas_limit(gas_limit: u64, info: &DispatchInfo) -> Result<(), BuiltInActorReason> {
+        let weight = info.weight;
+        if gas_limit < weight.ref_time() {
+            return Err(BuiltInActorReason::InsufficientGas);
+        }
+        Ok(())
+    }
+
+    fn dispatch_call(
+        origin: T::AccountId,
+        call: <T as Config>::RuntimeCall,
+        gas_limit: u64,
+    ) -> Result<(u64, Result<(), StakingErrorReason>), BuiltInActorReason> {
+        let call_info = call.get_dispatch_info();
+        Self::check_gas_limit(gas_limit, &call_info)?;
+        // Execute call
+        let res = call.dispatch(frame_system::RawOrigin::Signed(origin).into());
+        let actual_gas = extract_actual_weight(&res, &call_info).ref_time();
```

Comment:
There is a suggestion to charge for 5-15% more gas in these cases. It would allow to incentivize validators to execute built-in actors.  The second argument is that messages to built-in actor don't pass the usual execution flow so consume less gas.